--- src/parameterizedHeaderField.cpp	2005-06-03 13:09:11.000000000 +0200
+++ src/parameterizedHeaderField.cpp.oden	2011-03-23 08:20:03.114679609 +0100
@@ -82,12 +82,31 @@ void parameterizedHeaderField::parse(con
 	const string::value_type* const pstart = buffer.data() + position;
 	const string::value_type* p = pstart;
 
-	const string::size_type start = position;
+	string::size_type valueStart = position;
 
-	while (p < pend && *p != ';') ++p;
+	while (p < pend && parserHelpers::isSpace(*p))
+	{
+		++p;
+		++valueStart;
+	}
 
-	getValue().parse(buffer, start, position + (p - pstart));
+	// Advance up to ';', if any
+	string::size_type valueLength = 0;
+ 
+	while (p < pend && *p != ';')  // FIXME: support ";" inside quoted or RFC-2047-encoded text
+	{
+		++p;
+		++valueLength;
+	}
 
+	// Trim whitespaces at the end of the value
+	while (valueLength > 0 && parserHelpers::isSpace(buffer[valueStart + valueLength - 1]))
+		--valueLength;
+
+	// Parse value
+	getValue().parse(buffer, valueStart, valueStart + valueLength);
+
+	// Reset parameters
 	removeAllParameters();
 
 	// If there is one or more parameters following...
