--- src/mailbox.cpp	2005-03-18 22:26:47.000000000 +0100
+++ src/mailbox.cpp.oden	2011-03-23 08:17:01.893832869 +0100
@@ -84,6 +84,7 @@ void mailbox::parse(const string& buffer
 	// Temporary buffers for extracted name and address
 	string name;
 	string address;
+	bool hadBrackets = false;
 
 	while (p < pend)
 	{
@@ -276,6 +277,7 @@ void mailbox::parse(const string& buffer
 				}
 				else if (*p == '>')
 				{
+					hadBrackets = true;
 					break;
 				}
 				else if (!parserHelpers::isSpace(*p))
@@ -302,7 +304,7 @@ void mailbox::parse(const string& buffer
 
 	// Swap name and address when no address was found
 	// (email address is mandatory, whereas name is optional).
-	if (address.empty() && !name.empty())
+	if (address.empty() && !name.empty() && !hadBrackets)
 	{
 		m_email.empty();
 		m_email.reserve(name.size());
